File: Backend/ExpensePilot.API/ExpensePilot.API/Models/Transaction.cs

namespace ExpensePilot.API.Models
{
    public class Transaction
    {
        public int TransactionId { get; set; }
        public string UserId { get; set; }

        public int? CategoryId { get; set; }
        public TransactionType Type { get; set; }

        public decimal Amount { get; set; }

        public DateTime TransactionDate { get; set; }

        public string Title { get; set; }

        public string? Description { get; set; }

        //foriegn key
        public ApplicationUser? User { get; set; }
        public Category? Category { get; set; }

    }
    public enum TransactionType
    {
        Income,
        Expense
    }

}

File: AI/app/database/models/transaction.py
from sqlalchemy import (
    Column,
    Integer,
    String,
    Numeric,
    DateTime,
    Boolean,
    ForeignKey,
    Enum,
)
from sqlalchemy.orm import relationship
import enum

from app.database.base import Base


class TransactionType(enum.Enum):
    Income = "Income"
    Expense = "Expense"


class Transaction(Base):
    __tablename__ = "Transactions"

    transaction_id = Column("TransactionId", Integer, primary_key=True, index=True)
    user_id = Column("UserId", String, ForeignKey("AspNetUsers.Id"))

    category_id = Column(
    "CategoryId",
    Integer,
    ForeignKey("Categories.CategoryId"),
    nullable=True,
)

    type = Column("Type", Integer, nullable=False)

    amount = Column("Amount", Numeric(18, 2), nullable=False)
    transaction_date = Column("TransactionDate", DateTime, nullable=False)

    title = Column("Title", String, nullable=False)

    description = Column("Description", String, nullable=True)

    budget_exceeded = Column("BudgetExceeded", Boolean, default=False)



